pkg/apiserver/profiling/router.go

```diff
@@ -303,7 +311,29 @@ func (s *Service) viewSingle(c *gin.Context) {
 		_ = c.Error(err)
 		return
 	}
-	c.Data(http.StatusOK, "image/svg+xml", content)
+
+	// set default content-type for legacy profiling content.
+	contentType := "image/svg+xml"
+
+	if task.ProfileOutputType == ProfilingOutputTypeProtobuf {
```

Comment:
Prefer to explicitly add

```go
} else if task.ProfileOutputType == '' {
  contentType := "image/svg+xml"
}
```

instead of using fallbacks, since `task.ProfileOutputType == ''` is the only correct way to test whether it is a legacy record or not.